lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutCompletedEventDecoder.kt

```diff
@@ -0,0 +1,135 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.shopify.checkoutsheetkit
+
+import com.shopify.checkoutsheetkit.pixelevents.MoneyV2
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+
+@Serializable
+public data class CheckoutCompletedEvent(
+    public val orderDetails: OrderDetails? = null
+)
+
+@Serializable
+public data class OrderDetails(
+    public val id: String? = null,
+    public val email: String? = null,
+    public val phone: String? = null,
+    public val cart: CartInfo? = null,
+    public val paymentMethods: List<OrderPaymentMethod> = emptyList(),
+    public val billingAddress: Address? = null,
+    public val deliveries: List<DeliveryInfo> = emptyList(),
+)
+
+@Serializable
+public data class CartInfo(
+    public val token: String? = null, // TODO - is this sent?
+    public val lines: List<CartLine> = emptyList(),
+    public val price: PriceSet? = null,
+)
+
+@Serializable
+public data class PriceSet(
+    public val subtotal: MoneyV2? = null,
+    public val total: MoneyV2? = null,
+    public val taxes: MoneyV2? = null,
+    public val discounts: MoneyV2? = null,
+    public val shipping: MoneyV2? = null,
+)
+
+@Serializable
+public data class Address(
+    public val referenceId: String? = null,
+    public val firstName: String? = null,
+    public val lastName: String? = null,
+    public val address1: String? = null,
+    public val address2: String? = null,
+    public val city: String? = null,
+    public val countryCode: String? = null,
+    public val zoneCode: String? = null,
+    public val postalCode: String? = null,
+    public val phone: String? = null,
+    public val coordinates: ShippingCoordinates? = null
+)
+
+@Serializable
+public data class OrderPaymentMethod(
+    public val type: String? = null, // enum?
+    public val details: Map<String, String?>? = emptyMap(),
+)
+
+/**
+ * Current possible methods:
+ *  SHIPPING, PICK_UP, RETAIL, LOCAL, PICKUP_POINT, NONE
+ */
+@Serializable
+public data class DeliveryInfo(
+    public val method: String? = null,
```

Comment:
leaving this as a string rather than enum to avoid breakages when new values are added in web